tests_e2e/tests/multi_config_ext/multi_config_ext.py

```diff
@@ -0,0 +1,125 @@
+                                                                            resource_name="MCExt2")),
+            "MCExt3": MultiConfigExt.TestCase(VirtualMachineExtensionClient(self._context.vm,
+                                                                            VmExtensionIds.RunCommandHandler,
+                                                                            resource_name="MCExt3"))
+        }
+
+        # Add each extension to the VM and validate instance view has succeeded status.
+        # Each extension will be added with settings to echo its assigned test_guid:
+        # {
+        #     "source": {
+        #         "script": f"echo {t.test_guid}"
+        #     }
+        # }
+        log.info("")
+        log.info("Add 3 instances of RCv2 to the VM. Each instance will echo a unique guid...")
+        self.enable_extensions(test_cases=test_cases)
+        log.info("")
+        log.info("Check that each extension has the expected guid in its status message...")
+        self.assert_expected_guids_in_ext_status(test_cases=test_cases)
+
+        # Update MCExt3 with a new guid and add a new instance of RCv2 to the VM
+        updated_test_cases: Dict[str, MultiConfigExt.TestCase] = {
+            "MCExt3": MultiConfigExt.TestCase(VirtualMachineExtensionClient(self._context.vm,
+                                                                            VmExtensionIds.RunCommandHandler,
+                                                                            resource_name="MCExt3")),
+            "MCExt4": MultiConfigExt.TestCase(VirtualMachineExtensionClient(self._context.vm,
+                                                                            VmExtensionIds.RunCommandHandler,
+                                                                            resource_name="MCExt4"))
+        }
+        test_cases.update(updated_test_cases)
+
+        log.info("")
+        log.info("Update MCExt3 with new guid and add a new instance of RCv2 to the VM...")
+        self.enable_extensions(test_cases=updated_test_cases)
+        log.info("")
+        log.info("Check that each extension has the expected guid in its status message...")
+        self.assert_expected_guids_in_ext_status(test_cases=test_cases)
+
+        # Delete each extension on the VM and assert that there are no unexpected extensions left
+        log.info("")
+        log.info("Delete each instance of RCv2...")
+        for resource_name, test_case in test_cases.items():
+            log.info("")
+            log.info("Deleting {0} from the test VM".format(resource_name))
+            test_case.extension.delete()
+
+        vm: VirtualMachineClient = VirtualMachineClient(self._context.vm)
+        instance_view: VirtualMachineInstanceView = vm.get_instance_view()
+        if instance_view.extensions is not None and any(instance_view.extensions):
```

Comment:
updated so this test owns the vm

Reply:
why should it own the VM? I think the test needs to adapt to sharing the VM

Reply:
You're right, I think it is enough to just check that the name of the extension does not match the name of any extensions we've added for this test. I'll make the update

Reply:
yes, that would do it.... 

one thing, though, once an extension is added with a specific resource name (e.g. resource_name="CSE"), subsequent operations need to use the same name or CRP will give a really weird error. Maybe we should remove the resource_name parameter from VirtualMachineExtensionClient and set the resource name to be the same as the extension name when we call the SDK.

sharing VMs across multiple tests will for sure make writing those tests a little more complicated, but the trade-off is worth it, since creating a VM, adding a few extensions and then deleting the VM misses many scenarios that are real in prod VMs.

Reply:
updated